### ISSUE TITLE
We need more modularity and abstraction

### ISSUE THREAD

**Issue by astruxie** [EHanz/PackageStack#3]
Our current system is too unmodular. we need to abstract our processes and break up tasks. we shouldnt put more than 50 lines in a single file for readability.

**Comment by jkcarney** [comment]
I disagree. There are certain cases where I would prefer there to be 100+ lines of code in one file. Sometimes just everything should be in one file where possible. Because otherwise I'd have to tab between diferrent windows in my IDE and I dont know the keyboard shortcut for that

**Comment by astruxie** [comment]
> I disagree. There are certain cases where I would prefer there to be 100+ lines of code in one file. Sometimes just everything should be in one file where possible. Because otherwise I'd have to tab between diferrent windows in my IDE and I dont know the keyboard shortcut for that

i think you should be able to just click the other tab
i hope this helps!

best,
sam

**Comment by EHanz** [comment]
Hi Sam,
This is Evan the head of package stack. I saw you showed some concern in terms of abstraction and code allocation. I agree wholeheartedly with your idea that we should maintain a maximum of 50 lines for code files. However, if Mister Carney does so choose, I will grant him permission the usage of code files greater than 100 lines as long as another member of our organization gives it the once over and the green light. Remember that our aim and item of utmost importance is maintaining a package manager that is lightweight and readable! I hope this all helps and best of luck designing the future!
Excelsior,
Evan, CEO

**Comment by astruxie** [comment]
> Hi Sam, This is Evan the head of package stack. I saw you showed some concern in terms of abstraction and code allocation. I agree wholeheartedly with your idea that we should maintain a maximum of 50 lines for code files. However, if Mister Carney does so choose, I will grant him permission the usage of code files greater than 100 lines as long as another member of our organization gives it the once over and the green light. Remember that our aim and item of utmost importance is maintaining a package manager that is lightweight and readable! I hope this all helps and best of luck designing the future! Excelsior, Evan, CEO

Dear Evan,

thank you for the swift reply and your informative input! i do believe that your solution is fully obtainable and i will see that it is implemented in our work. although, i have concerns that Mister Carney's case will allow for our code to become non-streamlined and unconcise. as you stated in your reply, readability is of utmost importance at PackageStack and forgive me for disagreeing, but if Mister Carney is allowed usage of files greater than 100 lines (assuming he has the greenlight from another member of the team for quality assurance) then we will threaten this readability greatly, sir.

Godspeed,
Sam, Manager of technical stack asset team @ Package Stack Inc

**Comment by jkcarney** [comment]
> Hi Sam, This is Evan the head of package stack. I saw you showed some concern in terms of abstraction and code allocation. I agree wholeheartedly with your idea that we should maintain a maximum of 50 lines for code files. However, if Mister Carney does so choose, I will grant him permission the usage of code files greater than 100 lines as long as another member of our organization gives it the once over and the green light. Remember that our aim and item of utmost importance is maintaining a package manager that is lightweight and readable! I hope this all helps and best of luck designing the future! Excelsior, Evan, CEO

Dear Evan,

Thank you for this. I really appreciate you giving me the green light as to having 100+ lines in a singular file. I believe strongly there are certain cases it would benefit our design. One of our tenants is that it should be readable, and sometimes I believe that will be the way forward. I would be happy to get approval of another team member next time.

Regards
Josh Carney
Lead Object Oriented Programming Designer and Programmer
@ Package Stack Inc

**Comment by mikewilzn** [comment]
Sam,

Here at PackageStack we always recommend using gedit for maximum readability of your code.

Best wishes,
Mike Wilson
Vice President
Senior Lightweight Programmer
